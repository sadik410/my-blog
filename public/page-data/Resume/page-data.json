{"componentChunkName":"component---src-pages-resume-tsx","path":"/Resume/","result":{"data":{"site":{"siteMetadata":{"resume":{"experience":[{"icon":"fa fa-mobile","companyName":"Little Big Smala","duration":"Jan 2020 – Dec 2021","role":"Front end developer","roleDescription":"integrator, React developer , React-Native et React-Native-Web","task":" Integrating html5 / css3 PSD mockups, Develop React, React-Native and React-Native-Web components, Creating generic components in Storybook"},{"icon":"fa fa-rocket ","companyName":"Toolynk","duration":"AVR 2018 – Jan 2020","role":"Front end developer","roleDescription":"integrator, React developer , React-Native","task":"Integrating html5 / css3 PSD mockups,Develop React and React-Native components, Creating generic components"},{"icon":"fa fa-briefcase","companyName":"3dfordeco","duration":"MARCH 2016 – MARCH 2018","role":"Integrator & web-Designer (2D ,3D)","roleDescription":"integrator, React developer , React-Native et React-Native-Web","task":"Creation UX/UI des maquettes PSD, Integrating html5 / css3 PSD mockups ,CMS administration,Creation 3D model with 3DS max"}],"mySkills":["Javascript ES6","reactJs","react-native"," React-Native-Web","Redux","GIT","storybook","HTML","CSS","SASS","LESS","Styled Components"]}}}},"pageContext":{}},"staticQueryHashes":["1717071199","374561337","63159454"]}