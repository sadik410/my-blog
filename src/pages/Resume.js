import React from "react";

import Layout from "../components/layout";
import SEO from "../components/seo";
import Section from "../components/section/Section";

const Resume = () => {
  return (
    <Layout>
      <SEO title="Resume" />
      <Section title="Resume">
        <div className="container resume">
          <h4 className="title">Experience</h4>
          <div className="row">
            <div className="col-md-12">
              <div className="main-timeline">
                <div className="timeline">
                  <div className="timeline-icon">
                    <i className="fa fa-mobile" />
                  </div>
                  <div className="timeline-content">
                    <h3 className="title">
                      Little Big Smala <span>Jan 2020 – Dec 2021</span>
                    </h3>
                    <p className="description">
                      <p>
                        <span>Front end developer: </span>integrator, React
                        developer , React-Native et React-Native-Web
                      </p>
                      <span>task:</span> Integrating html5 / css3 PSD mockups,
                      Develop React, React-Native and React-Native-Web
                      components, Creating generic components in Storybook
                    </p>
                  </div>
                </div>
                <div className="timeline">
                  <div className="timeline-icon">
                    <i className="fa fa-rocket"></i>
                  </div>
                  <div className="timeline-content">
                    <h3 className="title">
                      Little Big Smala <span>Jan 2020 – Dec 2021</span>
                    </h3>
                    <p className="description">
                      <p>
                        <span>Front end web developer: </span>integrator, React
                        developer , React-Native et React-Native-Web
                      </p>
                      <span>task:</span> Integrating html5 / css3 PSD mockups,
                      Develop React and React-Native components, Creating
                      generic components
                    </p>
                  </div>
                </div>
                <div className="timeline">
                  <div className="timeline-icon">
                    <i className="fa fa-briefcase"></i>
                  </div>
                  <div className="timeline-content">
                    <h3 className="title">
                      3dfordeco <span>Jan 2020 – Dec 2021</span>
                    </h3>
                    <p className="description">
                      <p>
                        <span>Integrator & web-Designer (2D ,3D) : </span>
                        integrator, React developer , React-Native et
                        React-Native-Web
                      </p>
                      <span>task:</span> Creation UX/UI des maquettes PSD,
                      Integrating html5 / css3 PSD mockups ,CMS
                      administration,Creation 3D model with 3DS max
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </Section>
    </Layout>
  );
};

export default Resume;
