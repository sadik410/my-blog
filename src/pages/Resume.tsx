import * as React from "react";
import { graphql } from "gatsby";

import Layout from "../components/layout";
import SEO from "../components/seo";
import Section from "../components/section/Section";
import ResumeItem from "../components/resume/ResumeItem";
import { myContext } from "../context/Provider";

const Resume = ({ data }) => {
  const { activeColor } = React.useContext(myContext);

  return (
    <Layout>
      {/*       <SEO title="Resume" />
       */}{" "}
      <Section title="Resume">
        <div className="container resume">
          <h1 className="title">
            Ex<span style={{ color: activeColor }}>perience</span>
          </h1>
          <div className="row">
            <div className="col-md-12">
              <div className="main-timeline">
                {data?.site?.siteMetadata?.resume?.experience.map(
                  ({
                    icon,
                    companyName,
                    duration,
                    role,
                    roleDescription,
                    task,
                  }) => (
                    <ResumeItem
                      key={icon}
                      icon={icon}
                      companyName={companyName}
                      duration={duration}
                      role={role}
                      roleDescription={roleDescription}
                      task={task}
                    />
                  )
                )}
              </div>
            </div>
            <div className="my_skills col-md-12">
              <h1 className="my_skills__title">
                My
                <span style={{ color: activeColor }}> Skills</span>
              </h1>
              <div>
                <ul className="container  row justify-content-between flex-wrap">
                  {data?.site?.siteMetadata?.resume?.mySkills.map((name) => (
                    <li key={name}>{name}</li>
                  ))}
                </ul>
              </div>
            </div>
          </div>
        </div>
      </Section>
    </Layout>
  );
};

export default Resume;

export const query = graphql`
  query ResumePageQuery {
    site {
      siteMetadata {
        resume {
          experience {
            icon
            companyName
            duration
            role
            roleDescription
            task
          }
          mySkills
        }
      }
    }
  }
`;
