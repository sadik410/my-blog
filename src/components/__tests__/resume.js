import * as React from "react";
import { render } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import Resume from "../../pages/Resume";

const mySkills = [
  "Javascript ES6",
  "reactJs",
  "react-native",
  " React-Native-Web",
  "Redux",
  "GIT",
  "storybook",
  "HTML",
  "CSS",
  "SASS",
  "LESS",
  "Styled Components",
];
beforeAll(() => {
  useStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        title: `Gatsby Default Starter`,
        description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
        author: `@gatsbyjs`,
        resume: {
          experience: [
            {
              icon: "fa fa-mobile",
              companyName: "Little Big Smala",
              duration: "Jan 2020 – Dec 2021",
              role: "Front end developer",
              roleDescription:
                "integrator, React developer , React-Native et React-Native-Web",
              task:
                " Integrating html5 / css3 PSD mockups, Develop React, React-Native and React-Native-Web components, Creating generic components in Storybook",
            },
            {
              icon: "fa fa-rocket ",
              companyName: "Toolynk",
              duration: "AVR 2018 – Jan 2020",
              role: "Front end developer",
              roleDescription: "integrator, React developer , React-Native",
              task:
                "Integrating html5 / css3 PSD mockups,Develop React and React-Native components, Creating generic components",
            },
            {
              icon: "fa fa-briefcase",
              companyName: "3dfordeco",
              duration: "MARCH 2016 – MARCH 2018",
              role: "Integrator & web-Designer (2D ,3D)",
              roleDescription:
                "integrator, React developer , React-Native et React-Native-Web",
              task:
                "Creation UX/UI des maquettes PSD, Integrating html5 / css3 PSD mockups ,CMS administration,Creation 3D model with 3DS max",
            },
          ],
          mySkills,
        },
        panelColors: [],
      },
    }
  }));
});

afterAll(() => {
  useStaticQuery.mockRestore();
});

describe("Home", () => {
  it("renders correctly", () => {
    const { container } = render(<Resume />);
    expect(container).toMatchSnapshot();
  });
  it("render name correctly ", () => {
    const { getByText } = render(<Resume />);
    expect(
      getByText((content, element) => {
        return (
          element.tagName.toLowerCase() === "h1" &&
          element.textContent === "Experience"
        );
      })
    ).toBeInTheDocument();
  });

  /* it("render role correctly ", () => {
    const { getByText } = render(<Resume />);
    mySkills.forEach((item) => {
      const getTagLi = (content, element) => {
        return element.tagName.toLowerCase() === "li";
      };
      expect(getByText(getTagLi)).toBeInTheDocument();
    });
  }); */
});
