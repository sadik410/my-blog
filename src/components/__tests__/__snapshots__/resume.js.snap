// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="panel"
    >
      <button
        class="panel__switcher"
      >
        <i
          aria-hidden="true"
          class="fa fa-gear"
        />
      </button>
      <div
        class="panel__body "
      >
        <div
          class="panel__title"
        >
          Main Color
        </div>
        #2196F3
        <div
          class="panel__main-color d-flex justify-content-between flex-wrap"
        />
      </div>
    </div>
    <header
      style="padding-top: 20px; padding-bottom: 20px;"
    >
      <div
        class="header row justify-content-between"
      >
        <h1
          class="name"
        >
          Sadik
          <span
            style="color: rgb(33, 150, 243);"
          >
             Milliti
          </span>
        </h1>
        <nav>
          <ul
            class=""
          />
        </nav>
      </div>
    </header>
    <div>
      <main>
         
        <section
          class=" flex-column section"
        >
          <div
            class="section__header"
            style="background-color: rgb(33, 150, 243);"
          >
            <h2
              class="section__title"
            >
              Resume
            </h2>
          </div>
          <div
            class="section__body"
          >
            <div
              class="container resume"
            >
              <h1
                class="title"
              >
                Ex
                <span
                  style="color: rgb(33, 150, 243);"
                >
                  perience
                </span>
              </h1>
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="main-timeline"
                  />
                </div>
                <div
                  class="my_skills col-md-12"
                >
                  <h1
                    class="my_skills__title"
                  >
                    My
                    <span
                      style="color: rgb(33, 150, 243);"
                    >
                       Skills
                    </span>
                  </h1>
                  <div>
                    <ul
                      class="container  row justify-content-between flex-wrap"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </main>
      <footer
        style="margin-top: 2rem;"
      >
        Â© 
        2021
        , Built with
         
        <a
          href="https://www.gatsbyjs.com"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>
</div>
`;
